internal/dataplane/translator/backendref.go

```diff
@@ -23,42 +34,65 @@ func backendRefsToKongStateBackends(
 	for _, backendRef := range backendRefs {
 		logger := loggerForBackendRef(logger, route, backendRef)
 
-		if util.IsBackendRefGroupKindSupported(
-			backendRef.Group,
-			backendRef.Kind,
-		) && newRefChecker(backendRef).IsRefAllowedByGrant(allowed) {
-			port := int32(-1)
-			if backendRef.Port != nil {
-				port = int32(*backendRef.Port)
-			}
-			namespace := route.GetNamespace()
-			if backendRef.Namespace != nil {
-				namespace = string(*backendRef.Namespace)
-			}
-			backend, err := kongstate.NewServiceBackendForService(
-				k8stypes.NamespacedName{
-					Namespace: namespace,
-					Name:      string(backendRef.Name),
-				},
-				kongstate.PortDef{
-					Mode:   kongstate.PortModeByNumber,
-					Number: port,
-				},
-			)
-			if err != nil {
-				logger.Error(err, "failed to create ServiceBackend for backendRef")
-			}
-			if backendRef.Weight != nil {
-				backend.SetWeight(*backendRef.Weight)
+		nn := client.ObjectKey{
+			Name:      string(backendRef.Name),
+			Namespace: route.GetNamespace(),
+		}
+		if backendRef.Namespace != nil {
+			nn.Namespace = string(*backendRef.Namespace)
+		}
+
+		if backendRef.Kind == nil {
+			logger.Error(nil, "Object requested backendRef to target, but no Kind was specified, skipping...")
+			continue
+		}
+
+		var err error
+		switch *backendRef.Kind {
+		case "Service":
+			_, err = storer.GetService(nn.Namespace, nn.Name)
+		default:
+			err = fmt.Errorf("unsupported kind %q, only 'Service' is Supported", *backendRef.Kind)
+		}
+		if err != nil {
+			switch errors.As(err, &store.NotFoundError{}) {
+			case true:
+				logger.Error(nil, "Object requested backendRef to target, but it does not exist, skipping...")
+			case false:
+				logger.Error(nil, "Object requested backendRef to target, but no ReferenceGrant permits it, skipping...")
```

Comment:
I believe this message is now outdated - we log it when the `err != nil` where `err` might come either from getting the service or from an unsupported kind case. 
```suggestion
				logger.Error(err, "Object requested backendRef to target, but an error occurred, skipping...")
```